docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,66 @@
+
+To enable LTO add `--profile` option with LTO file path `tools\profiles\extensions\lto.json` to the build command.
+
+<span class="notes">**Note**: For profile extensions you have to put the full path relative to the project's root folder.</span>
+
+To enable LTO with the `release` profile:
+
+```
+mbed compile -t TOOLCHAIN -m TARGET --profile release --profile mbed-os/tools/profiles/extensions/lto.json
+```
+
+Example LTO profile memory savings for [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky):
+
+|Build type|Total static RAM memory (data + BSS)|Total flash memory (text + data)|
+| --- | --- | --- |
+| GCC_ARM - release - no LTO | 12,096B | 44,628B |
+| GCC_ARM - release - LTO | 11,800B | 41,088B |
+|***saved memory***|296B|3,540B|
+| ARM - release - no LTO | 10,365B | 35,496B |
+| ARM - release - LTO | 10,153B | 31,514B |
+|***saved memory***|212B|‭3,982‬B|
+
+LTO profile build results for [mbed-cloud-client-example](https://github.com/ARMmbed/mbed-cloud-client-example)
+
+|Build type|Total Static RAM memory (data + BSS)|Total Flash memory (text + data)|
+| --- | --- | --- |
+| GCC_ARM - release - no LTO | 59,760B | 389,637B |
+| GCC_ARM - release - LTO | 59,432B | 354,167B |
+|***saved memory***| 328B | ‭35,470‬B|
+| ARM - release - no LTO | 58,099B | 353,849B |
+| ARM - release - LTO | 57,150B | 322,500B |
+|***saved memory***| 949B | ‭31,349‬B|
+
+<span class="notes">**Note**: In LTO builds, the compiler produces bytecode/bitcode instead of regular object code. It's hard to analyze this output with object code analysis tools.</span>
+
+## Limitations
+
+- LTO slows down build process.
+- It’s very hard to control memory placement when using LTO.
+- LTO performs heavy memory optimizations that break debugging.
+- In LTO builds, the compiler produce bytecode/bitcode instead of regular object code. It's hard to analyze this output with object code analysis tools.
+- LTO could cause increases to the stack space needed due to cross-object inlining.
+
+### Arm
+
+- No bitcode libraries: armlink only supports bitcode objects on the command line. It does not support bitcode objects coming from libraries. armlink gives an error message if it encounters a file containing bitcode while loading from a library.
+- Partial Linking is not supported with LTO as it only works with elf objects not bitcode files.
+- Arm recommends that link time optimization is only performed on code and data that does not require precise placement in the scatter file, with general input section selectors such as *(+RO) and .ANY(+RO) used to select sections generated by link time optimization. It is not possible to match bitcode in .llvmbc sections by name in a scatter file.
+- Bitcode objects are not guaranteed to be compatible across compiler versions. This means that you should ensure all your bitcode files are built using the same version of the compiler when linking with LTO.
+
+### IAR
```

Comment:
We don't support IAR in the 6 docs; you can remove this bit.

Reply:
removed